### PR TITLE
documentation(EJ2-845801): committed maps new features in Volume 3.

### PR DESCRIPTION
**Pull request by indumathi-murali**

### CODE REVIEW

**Comment by indumathi-murali** [comment]
Tested sample link for label template -
JS sample - https://stackblitz.com/edit/pgss3a-twirup?file=index.js,index.html
TS sample - https://stackblitz.com/edit/9sanc1-lygqkn?file=index.ts,index.html

Tested sample link for line break -
JS sample - https://stackblitz.com/edit/pgss3a-1s6s36?file=index.js,index.html
TS sample - https://stackblitz.com/edit/9sanc1-8cshk3?file=index.ts,index.html

**Comment by indumathi-murali** [comment]
Sample link - Array binding example for table data
JS - https://stackblitz.com/edit/pgss3a-oxq3e6?file=index.js,index.html
TS - https://stackblitz.com/edit/9sanc1-9srbph?file=index.ts,index.html

**Comment by indumathi-murali** [comment]
Array Cell data- sample tested link
JS - https://stackblitz.com/edit/pgss3a-qfzh3x?file=index.js,index.html
TS - https://stackblitz.com/edit/9sanc1-cxwbnq?file=index.ts,index.html

**Comment by indumathi-murali** [comment]
Sample tested with JSON table binding data
JS - https://stackblitz.com/edit/pgss3a-xdz2hp?file=index.js,index.html
TS - https://stackblitz.com/edit/9sanc1-trokqf?file=index.ts,index.html

**Comment by indumathi-murali** [comment]
Sample tested - cell JSON binding
JS - https://stackblitz.com/edit/9sanc1-pwrthn?file=index.ts,index.html
TS - https://stackblitz.com/edit/9sanc1-pwrthn?file=index.ts,index.html

**Comment by sabari-senthamaraikannan** [comment]
> Tested sample link for label template - JS sample - https://stackblitz.com/edit/pgss3a-twirup?file=index.js,index.html TS sample - https://stackblitz.com/edit/9sanc1-lygqkn?file=index.ts,index.html
> 
> Tested sample link for line break - JS sample - https://stackblitz.com/edit/pgss3a-1s6s36?file=index.js,index.html TS sample - https://stackblitz.com/edit/9sanc1-8cshk3?file=index.ts,index.html

1. Center align the content within the label template. Use proper color in the circle shape.
2. Provide maxLabelLength in the y-axis to show all the characters. Remove the space in the labels.

**Comment by sabari-senthamaraikannan** [comment]
> Sample link - Array binding example for table data JS - https://stackblitz.com/edit/pgss3a-oxq3e6?file=index.js,index.html TS - https://stackblitz.com/edit/9sanc1-9srbph?file=index.ts,index.html

Ensure that the values in the template are properly in center. Use proper color in the circle shape.

**Comment by sabari-senthamaraikannan** [comment]
> Array Cell data- sample tested link JS - https://stackblitz.com/edit/pgss3a-qfzh3x?file=index.js,index.html TS - https://stackblitz.com/edit/9sanc1-cxwbnq?file=index.ts,index.html

Ensure that the values in the template are properly in center. Use proper color in the circle shape.

**Comment by sabari-senthamaraikannan** [comment]
> Sample tested with JSON table binding data JS - https://stackblitz.com/edit/pgss3a-xdz2hp?file=index.js,index.html TS - https://stackblitz.com/edit/9sanc1-trokqf?file=index.ts,index.html

Use value instead of the Year in the template content.

**Comment by sabari-senthamaraikannan** [comment]
> Sample tested - cell JSON binding JS - https://stackblitz.com/edit/9sanc1-pwrthn?file=index.ts,index.html TS - https://stackblitz.com/edit/9sanc1-pwrthn?file=index.ts,index.html

Title does not match with the content in the cell label template. Keep the accident label alone in the label template.

File: ej2-javascript/heatmap-chart/axis.md [hunk 1]
```diff
@@ -214,6 +214,37 @@ The text style of the axis labels can be customized using the following options
 {% previewsample "page.domainurl/code-snippet/heatmap/axis-cs13" %}
 {% endif %}
 
+### Providing line breaks
+
+Axis labels with line breaks improve the readability of the HeatMap by dividing the text labels on an axis into multiple lines. Line breaks are added to the axis labels by including the < br> character in the axis label content.
```
Comment (harivenkateshe) [inline ej2-javascript/heatmap-chart/axis.md:219]: Use the term "splitting" instead of "dividing".

**Comment by indumathi-murali** [comment]
> > Tested sample link for label template - JS sample - https://stackblitz.com/edit/pgss3a-twirup?file=index.js,index.html TS sample - https://stackblitz.com/edit/9sanc1-lygqkn?file=index.ts,index.html
> > Tested sample link for line break - JS sample - https://stackblitz.com/edit/pgss3a-1s6s36?file=index.js,index.html TS sample - https://stackblitz.com/edit/9sanc1-8cshk3?file=index.ts,index.html
> 
> 1. Center align the content within the label template. Use proper color in the circle shape.
> 2. Provide maxLabelLength in the y-axis to show all the characters. Remove the space in the labels.

Modified sample -
https://stackblitz.com/edit/pgss3a-2qiapc?file=index.js,index.html
https://stackblitz.com/edit/9sanc1-4ywzx7?file=index.ts,index.html

https://stackblitz.com/edit/pgss3a-dxhyps?file=index.js,index.html
https://stackblitz.com/edit/9sanc1-7q2rrl?file=index.ts,index.html

File: ej2-javascript/heatmap-chart/appearance.md [hunk 2]
```diff
@@ -333,6 +333,144 @@ The format of the data label, such as currency, decimal, percent etc. can be cha
 {% previewsample "page.domainurl/code-snippet/heatmap/appearance-cs10" %}
 {% endif %}
 
+## Data label template
```
Comment (harivenkateshe) [inline ej2-javascript/heatmap-chart/appearance.md:336]: 1. Title: "Template"
2. Move **all** the titles and sub-titles accordingly inside "Data label"

**Comment by indumathi-murali** [comment]
> > Sample link - Array binding example for table data JS - https://stackblitz.com/edit/pgss3a-oxq3e6?file=index.js,index.html TS - https://stackblitz.com/edit/9sanc1-9srbph?file=index.ts,index.html
> 
> Ensure that the values in the template are properly in center. Use proper color in the circle shape.

Modified sample tested link -
Array -table
JS - https://stackblitz.com/edit/pgss3a-mxs3f9?file=index.js
TS - https://stackblitz.com/edit/9sanc1-kpfavc?file=index.ts,index.html

**Comment by indumathi-murali** [comment]
> > Sample tested - cell JSON binding JS - https://stackblitz.com/edit/9sanc1-pwrthn?file=index.ts,index.html TS - https://stackblitz.com/edit/9sanc1-pwrthn?file=index.ts,index.html
> 
> Title does not match with the content in the cell label template. Keep the accident label alone in the label template.

Modified sample and tested link. Title is already changed. I have removed and keep the accident label alone.
JSON cell - 
JS - https://stackblitz.com/edit/pgss3a-qhc45q?file=index.js
TS - https://stackblitz.com/edit/9sanc1-pwrthn?file=index.ts,index.html

**Comment by indumathi-murali** [comment]
> > Sample tested with JSON table binding data JS - https://stackblitz.com/edit/pgss3a-xdz2hp?file=index.js,index.html TS - https://stackblitz.com/edit/9sanc1-trokqf?file=index.ts,index.html
> 
> Use value instead of the Year in the template content.

Modified Sample link . I have tried to give the value instead of Year the label is undefined. So i didn't change that.
Json table - 
JS - https://stackblitz.com/edit/pgss3a-bmkhn5?file=index.js
TS - https://stackblitz.com/edit/9sanc1-srsnkx?file=index.ts,index.html

File: ej2-javascript/heatmap-chart/axis.md (hunk 1 above)
Comment (sabari-senthamaraikannan) [reply ej2-javascript/heatmap-chart/axis.md]: The **"\<br>"** character is used to add line breaks to the axis labels.

**Review by sabari-senthamaraikannan — APPROVED** [review]

**Pull request merged**